packages/chain-controller/src/SnapChainProviderClient.test.ts

```diff
@@ -0,0 +1,72 @@
+import { ChainRpcMethod } from '@metamask/chain-api';
+import type { SnapId } from '@metamask/snaps-sdk';
+import { HandlerType } from '@metamask/snaps-utils';
+import type { Json } from '@metamask/utils';
+
+import { SnapChainProviderClient } from './SnapChainProviderClient';
+import { SnapHandlerClient } from './SnapHandlerClient';
+
+describe('SnapChainProviderClient', () => {
+  const handleRequest = jest.fn();
+
+  const snapId = 'local:localhost:3000' as SnapId;
+  const snapController = {
+    handleRequest,
+  };
+  const snapClient = new SnapHandlerClient({
```

Comment:
Similar to other comment, what are your thoughts on creating a new SnapHandlerClient for each test that needs one?